daemon/containerd/image.go

```diff
@@ -103,9 +104,22 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 			if i.UpdatedAt.After(lastUpdated) {
 				lastUpdated = i.UpdatedAt
 			}
+			if isDanglingImage(i) {
+				if len(tagged) > 1 {
```

Comment:
Doesn't this line only run if `tagged` is non-empty since it's inside a loop that's looping over `tagged`'s entries? :sweat_smile:

Reply:
(what I mean is that this `if`'s condition will always be true)

Reply:
It won't be true if tagged has only one element (which we are currently iterating) AND it's dangling.

If any image from tagged has the dangling name AND tagged is more than 1 element, then it means that the image itself is not dangling and there shouldn't be any image with the special dangling name.